### PR TITLE
devops(demo): support scaffol-stark-demo workflow

### PR DESCRIPTION
**Pull request by oceanlvr**
# Task name here

Workflows / vercel deployment / version control

The final outcome is to have a sample website out of scaffold-stark repo. There are some ways this can be done, we can just add one more workflow to it. This is a simpler approach. 
Or we can have a public repository called scaffol-stark-demo, that is like a website sample out of Scaffold-stark.
Everytime there is an update on the main repo, the workflow should update another scaffol-stark-demo repo and run a workflow on it to deploy it to vercel. 

## Types of change

- [x] Feature
- [ ] Bug
- [ ] Enhancement

## Comments (optional)

I choose use env to control the deployed block chain network. It might lead to a bit unmaintainable.

every time we push code to `nextjs` folder on the master branch, it would trigger this pipeline to deploy a new nextjs app on vercel. (using sepolia net)

### CODE REVIEW

**Comment by oceanlvr** [comment]
@jrcarlos2000  PTLA

File: .github/workflows/demo.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+name: scaffol-stark-demo workflow
```
Comment (0xquantum3labs) [inline .github/workflows/demo.yaml:1]: typo

File: .github/workflows/demo.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+          yarn install
+          vercel link --yes --project $VERCEL_PROJECT_ID --token $VERCEL_TOKEN --scope $VERCEL_ORG_ID
+          vercel --build-env NEXT_PUBLIC_IGNORE_BUILD_ERROR=true --prod --token $VERCEL_TOKEN --scope $VERCEL_ORG_ID
+
+      - name: Notify Slack on Success
+        if: success()
+        uses: slackapi/slack-github-action@v1.26.0
+        with:
+          channel-id: ${{ secrets.SLACK_CHANNEL_ID }}
+          slack-message: "GitHub build result: ${{ job.status }}\nRepository Name: ${{ github.repository }}\n${{ github.event.pull_request.html_url || github.event.head_commit.url }}"
```
Comment (0xquantum3labs) [inline .github/workflows/demo.yaml:52]:
```suggestion
          slack-message: "GitHub deployed to vercel result: ${{ job.status }}\nRepository Name: ${{ github.repository }}\n${{ github.event.pull_request.html_url || github.event.head_commit.url }}"
```

File: .github/workflows/demo.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        env:
+          SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+
+      - name: Notify Slack on Failure
+        if: failure()
+        uses: slackapi/slack-github-action@v1.26.0
+        with:
+          channel-id: ${{ secrets.SLACK_CHANNEL_ID }}
+          slack-message: "GitHub build result: ${{ job.status }}\nRepository Name: ${{ github.repository }}\n${{ github.event.pull_request.html_url || github.event.head_commit.url }}"
```
Comment (0xquantum3labs) [inline .github/workflows/demo.yaml:61]:
```suggestion
          slack-message: "GitHub deployed to vercel result: ${{ job.status }}\nRepository Name: ${{ github.repository }}\n${{ github.event.pull_request.html_url || github.event.head_commit.url }}"
```

**Review by 0xquantum3labs — CHANGES_REQUESTED** [review]

**Review by jrcarlos2000 — APPROVED** [review]

**Review by 0xquantum3labs — APPROVED** [review]

**Pull request merged**